scrapfly/api_response.py

```diff
@@ -565,3 +567,301 @@ def sink(self, path: Optional[str] = None, name: Optional[str] = None, file: Opt
             shutil.copyfileobj(file_content, f, length=131072)
 
         logger.info('file %s created' % file_path)
+
+
+class ScreenshotApiResponse:
+    def __init__(self, request: Request, response: Response, screenshot_config: ScreenshotConfig, api_result: Optional[bytes] = None):
+        self.request = request
+        self.response = response
+        self.screenshot_config = screenshot_config
+        self.result = self.handle_api_result(api_result)
+
+    @property
+    def image(self) -> Optional[str]:
+        binary = self.result.get('result', None)
+        if binary is None:
+            return ''
+
+        return binary
+
+    @property
+    def metadata(self) -> Optional[Dict]:
+        if not self.image:
+            return {}
+        
+        content_type = self.response.headers.get('content-type')
+        format = content_type[content_type.find('/') + 1:].split(';')[0]
+        format = 'jpg' if format == 'jpeg' else format # update as per the format api param
```

Comment:
you can remove the condition for jpg, the content-type must be iana compliant, otherwise it must be fixed on API side (the jpeg thing has been fixed since while)